docs/modules/mapstore/pages/configuring-a-generic-mapstore.adoc

```diff
@@ -373,13 +373,72 @@ mapConfig.setMapStoreConfig(mapStoreConfig);
 --
 ====
 
+|[[columns]]`type-name`
+|The type name of the compact GenericRecord. Use this property to map your record to an existing domain class.
+
+|
+The name of the map.
+|
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+  <map name="myMapName">
+    <map-store enabled="true">
+      <class-name>com.hazelcast.mapstore.GenericMapStore</class-name>
+      <properties>
+          <property name="data-connection-ref">my-mysql-database</property>
+          <property name="type-name">org.example.Person</property>
+      </properties>
+    </map-store>
+</hazelcast>
+----
+--
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  map:
+    mymapname:
+      map-store:
+        enabled: true
+        class-name: com.hazelcast.mapstore.GenericMapStore
+        properties:
+            data-connection-ref: my-mysql-database
+            type-name: org.example.Person
+----
+--
+Java::
++
+--
+[source,java]
+----
+MapConfig mapConfig = new MapConfig("myMapName");
+
+MapStoreConfig mapStoreConfig = new MapStoreConfig();
+mapStoreConfig.setClassName("com.hazelcast.mapstore.GenericMapStore");
+mapStoreConfig.setProperty("data-connection-ref", "my-mysql-database");
+mapStoreConfig.setProperty("type-name", "org.example.Person");
+
+mapConfig.setMapStoreConfig(mapStoreConfig);
+----
+--
+====
+
 |===
 
 == Supported backends
 
-You can use any database as the MapStore backend as long as you have its Hazelcast SQL Connector on the classpath.
+GenericMapStore needs a SQL Connector that supports `SELECT`, `UPDATE`, `SINK INTO` and `DELETE` statements.
```

Comment:
mongoDB is a noSQL does it support these statements 🤔

Reply:
SQL Connector here means Hazelcast SQL connector - https://docs.hazelcast.com/hazelcast/5.4/integrate/connectors#connectors-for-sql

The GenericMapStore produces SQL, which is run by Hazelcast SQL engine, translated into native queries of the particular database by the connector. Even for SQL databases (Postgres, MySQL, ...) the query is parsed and a different query may be generated (this is true especially for the `SINK INTO` which translates into some variant of UPSERT/MERGE statement.